subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -1048,3 +1048,40 @@ Alternatively, for Kotlin implementations, the Kotlin compiler can be configured
 include::sample[dir="snippets/configurationCache/samConversion/groovy",files="build.gradle[tags=configure]"]
 include::sample[dir="snippets/configurationCache/samConversion/kotlin",files="build.gradle.kts[tags=configure]"]
 ====
+
+[[config_cache:not_yet_implemented:accessing_top_level_at_execution]]
+=== Accessing top-level methods and variables of a build script at execution time
+
+Currently, Gradle cannot run a task restored from the configuration cache if a task action calls a method defined directly in the build script.
+
+In case of the Groovy DSL, the action fails when running from the cache with an error message about method not being found.
```

Comment:
```suggestion
For builds scripts written in Groovy, the task fails because the method cannot be found.
```

Suggestion: rephrase introductory sentence to better scope the situation. Rephrase "the action fails when running from the cache" to "the task fails" since we haven't clearly defined what "action" means and if the action fails, the task fails as well (I think?). If the task fails, it stands to reason that there's an error message, so we can leave that out and just state the cause instead.